### PR TITLE
Rework help-box making it reusable and more maintenable by regrouping markups

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Help box was wrong used in some pages, I also extracted it to a reusable component. This aim the fact that we have only one file to change if we want to change the help box markup, it's also a POC about reusable components we should have on the project
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | Build assets then try to see these helps box on every BO pages, they should work as expected

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18184)

### CODE REVIEW

File: src/PrestaShopBundle/Resources/views/Admin/Common/HelpBox/helpbox.html.twig [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *#}
+
+ <span 
+   class="help-box{% if classes is defined %} {{ classes }}{% endif %}"
+   data-toggle="popover"
+   {% if content is defined %}
+    data-content="{{ content }}"
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/Common/HelpBox/helpbox.html.twig]: one space indent? :sweat_smile:

Comment (NeOMakinG) [reply src/PrestaShopBundle/Resources/views/Admin/Common/HelpBox/helpbox.html.twig]: wtf ahah

**Review by PierreRambaud — DISMISSED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/sorting.html.twig [hunk 2]
```diff
@@ -27,10 +27,7 @@
     <div class="col-lg-6">
       <div class="module-sorting-search-wording">
         <span id="selected_modules" class="module-search-result-title module-search-result-wording">{{ '%nbModules% modules and services selected for you'|trans({'%nbModules%' : totalModules}, 'Admin.Modules.Feature') }}</span>
-        <span class="help-box" data-toggle="popover"
-          data-title="{{ "Selection"|trans({}, 'Admin.Modules.Feature') }}"
-          data-content="{{ "Customize your store with this selection of modules recommended for your shop, based on your country, language and version of PrestaShop. It includes the most popular modules from our Addons marketplace, and free partner modules."|trans({}, 'Admin.Modules.Help') }}">
-        </span>
+        {% include '@Common/HelpBox/helpbox.html.twig' with { 'content' : "Selection"|trans({}, 'Admin.Modules.Feature')  , 'title' : "Customize your store with this selection of modules recommended for your shop, based on your country, language and version of PrestaShop. It includes the most popular modules from our Addons marketplace, and free partner modules."|trans({}, 'Admin.Modules.Help')   } %}
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/sorting.html.twig]:
```suggestion
        {% include '@Common/HelpBox/helpbox.html.twig' with { 'content' : "Selection"|trans({}, 'Admin.Modules.Feature'), 'title' : "Customize your store with this selection of modules recommended for your shop, based on your country, language and version of PrestaShop. It includes the most popular modules from our Addons marketplace, and free partner modules."|trans({}, 'Admin.Modules.Help') } %}
```

File: src/PrestaShopBundle/Resources/views/Admin/Module/updates.html.twig [hunk 3]
```diff
@@ -36,10 +36,8 @@
     <div class="col-lg-10">
       <div id="module-short-list-update" class="module-short-list">
         <span class="module-search-result-title module-search-result-wording">{{ '%nbModules% modules to update'|trans({'%nbModules%' : modules|length}, 'Admin.Modules.Feature') }}</span>
-        <span class="help-box" data-toggle="popover"
-          data-title="{{ "Modules to update"|trans({}, 'Admin.Modules.Feature') }}"
-          data-content="{{ "Update these modules to enjoy their latest versions."|trans({}, 'Admin.Modules.Help') }}">
-        </span>
+        {% include '@Common/HelpBox/helpbox.html.twig' with { 'content' :  "Update these modules to enjoy their latest versions."|trans({}, 'Admin.Modules.Help') , 'title' :  "Modules to update"|trans({}, 'Admin.Modules.Feature')  } %}
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Module/updates.html.twig]:
```suggestion
        {% include '@Common/HelpBox/helpbox.html.twig' with { 'content' : "Update these modules to enjoy their latest versions."|trans({}, 'Admin.Modules.Help'), 'title' : "Modules to update"|trans({}, 'Admin.Modules.Feature') } %}
```

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Progi1984** [comment]
@NeOMakinG Don't forget to create issue for your PR.

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

I found an issue related to help box
In the BO => Product page => edit a product with combination => Go to the Combinations Tab => Edit a combinations => click on the help boxes => click on "Go to product" without closing help boxes => see error
image

This issue is supposed to be fixed in this PR?
Thanks!

**Comment by NeOMakinG** [comment]
It looks not related to this PR, I think it does the same thing on every branches, my PR doesnt change any behavior :/

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

It is OK :heavy_check_mark: 

Thanks!

**Comment by khouloudbelguith** [comment]
@NeOMakinG, if this PR is merged, can we close this issue: https://github.com/PrestaShop/PrestaShop/issues/18759?

Thanks!

**Comment by PierreRambaud** [comment]
Thanks @NeOMakinG

**Comment by NeOMakinG** [comment]
@khouloudbelguith yes it can be closed, Thanks :)

**Comment by Progi1984** [comment]
Done

**Pull request merged**